### PR TITLE
Feat/ethers

### LINKED ISSUE #50: [BUG][0.3.0-alpha4] Publish error
Nautilus Version: `0.3.0-alpha4`

Error:
```bash
ReferenceError: _interrupt is not defined
```
Does not work for any tested dataset AND algorithm.
Same problem with OCEAN token and EUROe
Tested FileType URL in combinations:

1. FREE/ACCESS
2. FREE/COMPUTE
3. FIXED/ACCESS
4. FIXED/COMPUTE

Code:

```ts

async function main() {
    Nautilus.setLogLevel(LogLevel.Verbose)
    const nautilus = await Nautilus.create(web3, genx_config )

    await publishDataset(nautilus)
}

async function publishDataset(nautilus) {
    const pricingConfig: PricingConfigWithoutOwner = {
        type: 'free'
    }
    // const pricingConfig: PricingConfigWithoutOwner = {
    //     type: 'fixed',
    //     freCreationParams: {
    //         fixedRateAddress: genx_config.fixedRateExchangeAddress, 
    //         baseTokenAddress: '0xe974c4894996E012399dEDbda0bE7314a73BBff1', 
    //         baseTokenDecimals: 6,
    //         datatokenDecimals: 18, 
    //         fixedRate: '1', 
    //         marketFee: '0', 
    //         marketFeeCollector: '0x0000000000000000000000000000000000000000' 
    //     }
    // }

    // Create a new 'access' type service serving 'url' files
    const serviceBuilder = new ServiceBuilder(ServiceTypes.ACCESS, FileTypes.URL)

    const urlFile: UrlFile = {
        type: 'url', 
        url: 'https://bitcoin.org/bitcoin.pdf', 
        method: 'GET'
    }
    const name = 'My Datatoken Name'
    const symbol = 'SYMBOL'

    const service = serviceBuilder
        .setServiceEndpoint(genx_config.providerUri) 
        .setTimeout(0) 
        .addFile(urlFile)
        .setPricing(pricingConfig)
        .setDatatokenNameAndSymbol(name, symbol)
        .build()

    const assetBuilder = new AssetBuilder()
    const asset = assetBuilder
        .setType('dataset') 
        .setName('My New Asset')
        .setDescription('A publish asset building test') 
        .setAuthor('testAuthor')
        .setLicense('MIT') 
        .addService(service)
        .setOwner(web3.defaultAccount)
        .build()


    const result = await nautilus.publish(asset)
    console.log(result)

}
```

### LINKED ISSUE #52: [BUG][0.3.0-alpha4] access error
Nautilus Version: `0.3.0-alpha4`

Error: 
```bash
TypeError: Cannot read properties of undefined (reading '1')
```

Code:
```ts

async function main() {
    Nautilus.setLogLevel(LogLevel.Verbose)
    const nautilus = await Nautilus.create(web3, genx_config )
    console.log(nautilus.getOceanConfig())

    await access(nautilus)
}

async function access(nautilus) {

    const accessUrl = await nautilus.access({
        assetDid: 'did:op:8ca612cfc6ffac030ca1a0f3d63f2ba72ec6e409d487f19a815b43bc11160b34'
    })
    console.log(accessUrl)
}
```

### PR DESCRIPTION
**Pull request by moritzkirstein**
Fixes #52 
Fixes #50 

## Proposed Changes

  - Move to ocean.js 3.0
  - Refactor to use ethers over Web3

### CODE REVIEW

File: test/fixtures/Web3.ts
Comment (Abrom8) [inline test/fixtures/Web3.ts:1]: suggestion: rename the file to `Ethers.ts` or maybe `Signer.ts`

File: README.md [hunk 1]
```diff
@@ -156,25 +161,50 @@ const urlFile = {
...
+```ts
+// Example of a fixed asset
+serviceBuilder.setPricing({
+  type: 'fixed', // 'fixed' or 'free'
+  // freCreationParams can be ommitted for 'free' pricing schemas
+  freCreationParams: {
+    fixedRateAddress: '0x25e1926E3d57eC0651e89C654AB0FA182C6D5CF7', // Fixed Rate Contract address on Mumbai network
+    baseTokenAddress: '0xd8992Ed72C445c35Cb4A2be468568Ed1079357c8', // OCEAN token contract address on Mumbai network
+    baseTokenDecimals: 18, // adjusted to OCEAN token
+    datatokenDecimals: 18, // adjusted to OCEAN token
```
Comment (Abrom8) [inline README.md:194]: todo: I think it's not referring to the OCEAN token.

**Review by Abrom8 — CHANGES_REQUESTED** [review]

File: scripts/release-it-documentation-checker-plugin.mjs [hunk 2]
```diff
@@ -11,7 +11,14 @@ export default class DocsPlugin extends Plugin {
 
   async beforeBump() {
     const { version } = this.config.contextOptions
-    await this.prepareDocsForNewVersion(version)
+    const isDryRun = this.config.options['dry-run']
+    const withDocsOption = this.config.options['with-docs']
+
+    const shouldPrepareDocs =
+      withDocsOption === undefined || withDocsOption !== 'false'
```
[the remark this answers is not in the document]
Comment (moritzkirstein) [reply scripts/release-it-documentation-checker-plugin.mjs:18]: Yes, we want to always create docs unless the user specifically ops out of this option.

File: test/access.test.ts [hunk 3]
```diff
@@ -1,24 +1,25 @@
...
-    })
+    assert(false, 'Re-activate test')
+    // const fileUrl = await nautilus.access({
+    //   assetDid
+    // })
 
-    LoggerInstance.log(fileUrl)
-    assert(fileUrl)
+    // LoggerInstance.log(fileUrl)
+    // assert(fileUrl)
```
[the remark this answers is not in the document]
Comment (moritzkirstein) [reply test/access.test.ts]: Good catch. These all have been moved to `test/integration/` respectively. I removed the unused files.

File: README.md (hunk 1 above)
Comment (moritzkirstein) [reply README.md:189]: should be correct, see: https://github.com/oceanprotocol/contracts/blob/85a130307c7846cf354d8f919a52e0152d834906/addresses/address.json#L4C32-L4C32

**Review by Abrom8 — APPROVED** [review]

**Pull request merged**